### PR TITLE
Add multiplexer to boot pin

### PR DESCRIPTION
**Pull request by napowderly**
Screenshot 2024-03-12 at 1 44 44 PM

Using this MUX from TI: https://www.ti.com/lit/ds/symlink/sn74lvc1g3157.pdf?ts=1710189618915&ref_url=https%253A%252F%252Fwww.google.com%252F

### CODE REVIEW

File: elec/src/micro.ato [hunk 1]
```diff
@@ -47,4 +59,37 @@ module Micro:
...
+
+    input_b2 = new Pair
+    output.io ~ pin 3
+    output.gnd ~ power.gnd
+
+    output = new Pair
+    output.io ~ pin 4
+    output.gnd ~ power.gnd
+
+    control = new Pair
```
Comment (mawildoer) [inline elec/src/micro.ato:96]: Isn't this called select on the IC?

File: elec/src/micro.ato [hunk 2]
```diff
@@ -24,15 +24,27 @@ module Micro:
...
-    # Motor control
+    # CAN RX and Bootpin are shared, need to be multiplexed externally (backup for SW boot option)
+    button ~ mux.input_b1
+    can_tranciever.can_ttl.rx ~ mux.input_b2
+    button ~ micro.PA4 # for enumeration or something while micro is running
+    micro.PB8_BOOT0 ~ mux.output
+    mux.control ~ micro.PA6
+
+    # Pulldown reistor to default to boot selction - assert high with PA6 to enable CAN
+    select_resistor.1 ~ mux.control.io
```
Comment (mawildoer) [inline elec/src/micro.ato:44]: nit; flip this around so that it reads high-to-low voltage

File: elec/src/micro.ato [hunk 3]
```diff
@@ -24,15 +24,27 @@ module Micro:
...
     spi ~ micro.spi2 # (PB15 mosi, PB14 miso, PB13 sck, PB12 cs)
-    can_tranciever.can_ttl ~ micro.can_ttl # (PA12 tx, PA11 rx)
     can ~ can_tranciever.can
 
-    # Motor control
+    # CAN RX and Bootpin are shared, need to be multiplexed externally (backup for SW boot option)
+    button ~ mux.input_b1
+    can_tranciever.can_ttl.rx ~ mux.input_b2
+    button ~ micro.PA4 # for enumeration or something while micro is running
+    micro.PB8_BOOT0 ~ mux.output
```
Comment (mawildoer) [inline elec/src/micro.ato]: nit; flip this around output-to-input

File: elec/src/micro.ato (hunk 2 above)
Comment (mawildoer) [inline elec/src/micro.ato:44]: There aren't params set for this resistor yet

File: elec/src/micro.ato (hunk 1 above)
Comment (napowderly) [reply elec/src/micro.ato:96]: Yeah, can update for clarity.

File: elec/src/micro.ato (hunk 3 above)
Comment (napowderly) [reply elec/src/micro.ato]: Done

**Pull request merged**